add Documentation for Date and Time Picker (#199)
eclipsesource/jsonforms2-website: content/docs/date-time-picker.mdx

---
id: date-time-picker
title: Date and Time Picker

> ---

Reviewer: Add a description to this page

---
id: date-time-picker

> title: Date and Time Picker

Reviewer 2: ```suggestion
title: Date and Time Pickers
```

description: Configure Date and Time Picker
---

import { OneOfLabel, PropertyLabel, SchemaLabel, UiSchemaLabel } from '../../src/components/docs/labels'

import {
  TimeSchema,
  TimeUiSchema,
  TimeUiOptionSchema,
  DateSchema,
  DateUiSchema,
  DateUiSchemaOptions,
  DateTimeSchema,
  DateTimeUiSchema,
  DateTimeUiOptionSchema,
  TimeOptionTable,
  DateOptionTable,
  DateTimeOptionTable
} from '../../src/components/docs/date-time-picker';

> JSON Forms supports Time and Date Picker as well as a Date-Time Picker, that combines both functionalities. Options for customizing the Date/Time Picker are available for the React Material Renderer.

Reviewer 2: Let's cut to the chase: The reason we support all of them is because they are defined in the JSON Schema specification.
```suggestion
JSON Forms supports JSON Schema's "date", "time" and "date-time" formats. Additional options to customize the "date", "time" and "date-time" pickers are offered for the React Material renderer set.
```

## Time Picker

> There are two ways to integrate the Time Picker. It can be embedded via the schema or the UI schema. The Time Picker supports a variety of options, that can be configured in the UI schema.

Reviewer 2: "integrate" is not ideal as we're not really integrating something. We have the tester based system and the question is which is allowed to render. So we usually use "use", "selected", "triggered" etc.
```suggestion
The time picker will be used when `format: "time"` is specified for a string property in the JSON Schema. Alternatively `format: "time"` can also be specified via the UI Schema options.
```

### Schema Based

> An Input Field will be displayed as Time Picker by setting the format of the corresponding field to “time” in the schema.

Reviewer 2: "Input Field" is not really a name so it should be lower cases. We only upper case "JSON Schema" and "UI Schema" when used as names. Same with "Time Picker" (headlines are fine).

```suggestion
A control will be rendered as a time picker when the format of the corresponding string property is set to "time" in the JSON Schema.
```

> An Input Field will be displayed as Time Picker by setting the format of the corresponding field to “time” in the schema.

Reviewer 2: There are some weird quotations marks here

<TimeSchema />

### UI Schema Based

> An Input Field will also be displayed as Time Picker by setting the property "format" to "time" in the options in the UI schema.

Reviewer 2: ```suggestion
A string control will also be rendered as a time picker by setting the property "format" to "time" in the UI Schema options.
```

<TimeUiSchema />

### Options
An overview over the options can be seen in the example and the table below. Please note, that these options are only available for the Material Ui Renderer Set.
<TimeUiOptionSchema/>

> <TimeOptionTable />

Reviewer 2: Please describe a bit what the example is showcasing. Note that my suggested example text below is not how the example currently looks like.

```suggestion
### Options
The React Material renderer set offers additional UI Schema options to customize the appearance of the time picker text input as well as the picker itself.
<TimeOptionTable />

The following example showcases some of the options.
The text input is configured to only show the full hours while both hours and minutes are saved into the data.
The picker presents itself in `am/pm` format.
<TimeUiOptionSchema/>

```
